controllers/kubeletconfig_controller.go

```diff
@@ -0,0 +1,200 @@
+	rteconfig "github.com/openshift-kni/numaresources-operator/rte/pkg/config"
+	"github.com/openshift-kni/numaresources-operator/rte/pkg/sysinfo"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	kubeletConfigRetryPeriod = 30 * time.Second
+)
+
+// KubeletConfigReconciler reconciles a KubeletConfig object
+type KubeletConfigReconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Recorder  record.EventRecorder
+	Namespace string
+}
+
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=*
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=kubeletconfigs,verbs=get;list;watch
+
+func (r *KubeletConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.V(3).InfoS("Starting KubeletConfig reconcile loop", "object", req.NamespacedName)
+	defer klog.V(3).InfoS("Finish KubeletConfig reconcile loop", "object", req.NamespacedName)
+
+	nname := types.NamespacedName{
+		Name: defaultNUMAResourcesOperatorCrName,
+	}
+	instance := &nropv1alpha1.NUMAResourcesOperator{}
+	err := r.Get(ctx, nname, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// so we need to have a NUMAResourceOperator instance to be able to set owner refs
+			// correctly. And we just wait forever until it comes, which is expected to be "soon".
+			return ctrl.Result{RequeueAfter: kubeletConfigRetryPeriod}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// KubeletConfig changes are expected to be sporadic, yet are important enough
+	// to be made visible at kubernetes level. So we generate events to handle them
+
+	cm, err := r.reconcileConfigMap(ctx, instance, req.NamespacedName)
+	if err != nil {
+		klog.ErrorS(err, "failed to reconcile configmap", "controller", "kubeletconfig")
+
+		msg := fmt.Sprintf("Failed to update RTE config from kubelet config %s/%s", req.NamespacedName.Namespace, req.NamespacedName.Name)
+		r.Recorder.Event(instance, "Warning", "ProcessFailed", msg)
+		return ctrl.Result{}, err
+	}
+
+	msg := fmt.Sprintf("Updated RTE config %s/%s from kubelet config %s/%s", cm.Namespace, cm.Name, req.NamespacedName.Namespace, req.NamespacedName.Name)
+	r.Recorder.Event(instance, "Normal", "ProcessOK", msg)
+	return ctrl.Result{}, nil
+}
+
+func (r *KubeletConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&mcov1.KubeletConfig{}).
+		Owns(&corev1.ConfigMap{}).
+		Complete(r)
+}
+
+func (r *KubeletConfigReconciler) reconcileConfigMap(ctx context.Context, instance *nropv1alpha1.NUMAResourcesOperator, kcKey client.ObjectKey) (*corev1.ConfigMap, error) {
+	mcoKc := &mcov1.KubeletConfig{}
```

Comment:
Do we know for certain that there's only one?

Reply:
zero or one, yes. `KubeletConfig` is cluster scoped, so it must be unique by name.